### PR TITLE
Go Runtime with go mod 

### PR DESCRIPTION
**Pull request by thspinto**
## ☕ Purpose

Implement Go runtime with go mod as dependency manager.

## ❓ Questions

* Should we create a new runtime for go due to breaking changes?
* I used golang as name for the run time, but not sure if it is the best name.
* Where to host the official docker image?

## 🧐 Checklist

- [x] Add CI integration
- [x] Send to stable dir

## 🍩 Further details

Let me know what you think of this PR, I'll finish de CI integration after discussing the questions.

### CODE REVIEW

**Comment by thspinto** [comment]
# Related Issues

* https://github.com/kubeless/kubeless/issues/1136
* https://github.com/kubeless/kubeless/issues/1100
* https://github.com/kubeless/runtimes/issues/70

**Comment by thspinto** [comment]
> Hope that makes sense to you!

It does. I'm going to fix the tests and add the instructions to run the legacy version then.

Thanks @andresmgot.

**Comment by thspinto** [comment]
Done. Ready for review.

File: stable/golang/golang_legacy.jsonnet [hunk 1]
```diff
@@ -0,0 +1,37 @@
+{
```
Comment (andresmgot) [inline stable/golang/golang_legacy.jsonnet:1]: this file should contain the runtimes for `go1.10`, `go1.11` and `go1.12` :)

Comment (thspinto) [reply stable/golang/golang_legacy.jsonnet:1]: Lol. Missed that, thanks =).

**Comment by andresmgot** [comment]
Thanks for the changes! Do you mind updating the kubeless docs with the new changes here: https://github.com/kubeless/kubeless/blob/master/docs/runtimes.md#go?

**Comment by thspinto** [comment]
> Thanks for the changes! Do you mind updating the kubeless docs with the new changes here: https://github.com/kubeless/kubeless/blob/master/docs/runtimes.md#go?

Sure, I'll send a PR over there! Thanks!

**Pull request merged**